modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java

```diff
@@ -1454,11 +1491,8 @@ private void _validateIndexed(
 			throw new ObjectFieldDBTypeException("Blob type is not indexable");
 		}
 
-		if (((!Objects.equals(
-				businessType, ObjectFieldConstants.BUSINESS_TYPE_ATTACHMENT) &&
-			  !Objects.equals(dbType, ObjectFieldConstants.DB_TYPE_CLOB) &&
-			  !Objects.equals(dbType, ObjectFieldConstants.DB_TYPE_STRING)) ||
-			 indexedAsKeyword) &&
+		if (!_isObjectFieldIndexedAsText(
+				businessType, dbType, indexed, indexedAsKeyword) &&
```

Comment:
@devpedrotavares with your changes, this condition will be `true` if:

```
(!indexed || indexedAsKeyword ||
(!Objects.equals(businessType, ObjectFieldConstants.BUSINESS_TYPE_ATTACHMENT) &&
!Objects.equals(dbType, ObjectFieldConstants.DB_TYPE_CLOB) &&
!Objects.equals(dbType, ObjectFieldConstants.DB_TYPE_STRING))) &&
!Validator.isBlank(indexedLanguageId)
```
therefore, it will be true if:

```
!indexed && !Validator.isBlank(indexedLanguageId)
```
In this case we will throw:

```
throw new ObjectFieldDBTypeException(
	"Indexed language ID can only be applied with type \"Clob\" " +
		"or \"String\" that is not indexed as a keyword");
```

This does not seem right to me.

Reply:
Okay, I updated the PR to keep the same behavior for this part.